tests/unit/cocoon/test_cocoon_printer.py

```diff
@@ -363,6 +383,33 @@ def test_format_holdings_response_success(
             for index, row in err.iterrows()
         ]
 
+    @parameterized.expand(
+        [
+            (
+                "standard_response",
+                responses,
+                2,
+                {"succ": ["ID00002", "ID00002"], "err": ["not found", "not found"]},
+            ),
+            ("empty_response", empty_response_with_full_shape, 0, {}),
+        ]
+    )
+    def test_format_reference_portfolios_response_success(
+        self, _, response, num_items, expected_value
+    ):
+        succ, err = format_reference_portfolios_response(response)
+        self.assertEqual(num_items, len(succ))
```

Comment:
I think  you can appease sonar by pulling the assertions into a new function, something like

`def assertResponses(self, succ, errr):`